compiler/src/dotty/tools/dotc/transform/localopt/ConstantFold.scala

```diff
@@ -76,9 +77,6 @@ import Simplify.desugarIdent
     //       cond.select(defn.Boolean_&&).appliedTo(elsep)
     //   the other case ins't handled intentionally. See previous case for explanation
 
-    case If(t @ Select(recv, _), thenp, elsep) if t.symbol eq defn.Boolean_! =>
```

Comment:
I'm confused, in the PR description you say that `if(!e) { a } else { b }` now `if(e) { b } else { a }`, but it seams that you just removed the case doing that... Could you had a test for this example?

Reply:
Hi, I looked into it: you are right. It's an error on my part, I didn't add the feature, I only added documentation for it. 

Sorry or the confusion...